pyproject.toml

```diff
@@ -1,7 +1,97 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab>=3,<=4", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.5.0", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupytext"
+description = "Jupyter notebooks as Markdown documents, Julia, Python or R scripts"
+license = { file = "LICENSE" }
+authors = [
+    { name = "Marc Wouts", email = "marc.wouts@gmail.com" },
+]
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "nbformat",
+    "pyyaml",
+    "toml",
+    "markdown-it-py>=1.0.0",
+    "mdit_py_plugins",
+]
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/mwouts/jupytext"
+source = "https://github.com/mwouts/jupytext"
+documentation = "https://jupytext.readthedocs.io/en/latest/index.html"
+
+[project.optional-dependencies]
+extension = [
+    "jupytext_extension",
+]
+dev = [
+    "autopep8",
+    "black",
+    "isort",
+    "flake8",
+    "pytest",
+    "pytest-cov>=2.6.1",
+    "gitpython",
+    "jupyterlab",
+    "notebook",
+    "nbconvert",
+    "ipykernel",
+]
+# left for back-compatibility
+rst2md = [
+    "sphinx-gallery~=0.7.0",
+]
+
+[project.scripts]
+"jupytext" = "jupytext.cli:jupytext"
+"jupytext-config" = "jupytext.jupytext_config:main"
+
+[tool.hatch.version]
+path = "jupytext/version.py"
```

Comment:
If the version is made dynamic, delete the static one and make a `_version.py` generated file.

Reply:
From my understanding, version is set manually in `version.py` file. So, `hatch` will read the version string from the file and use it for the package. If we would like to have a more dynamic versioning, we can look into [hatch-vcs](https://github.com/ofek/hatch-vcs) which is like `versioneer` for `hatch` build system.

Reply:
Yes currently we do set the version number in `jupytext/version.py`. I am ok with doing otherwise but I don't want to keep @mahendrapaipuri busy with this too long!